### PR TITLE
chore(classifier): retrain kernel cve impact model with refreshed patch data

### PR DESCRIPTION
**Pull request by superbuggy**
## What

Retrains the kernel CVE impact classifier against freshly scraped patch data. Updates all pipeline artifacts under `kernel-cve-impact-classifier/`: training/test CSVs, SMOTE-balanced dataset, CVSS/CWE cache, the XGBoost model, and test-result outputs (predictions, summary JSON, plot).

## Why

The training corpus is built from live kernel git data; periodic retrains keep the model current as new CVEs and fix-commits land upstream. This retrain improves overall accuracy (73.9% → 75.0%), precision (72.1% → 76.6%), recall (65.5% → 66.8%), and F1 (67.1% → 69.6%), with MODERATE-class recall rising from 44.4% to 48.1%.

| Metric | Previous (HEAD) | Current | Delta |
|--------|----------------|---------|-------|
| Accuracy | 73.9% | **75.0%** | +1.1pp |
| Precision | 72.1% | **76.6%** | +4.5pp |
| Recall | 65.5% | **66.8%** | +1.2pp |
| F1 | 67.1% | **69.6%** | +2.5pp |

Per-class recall also shifted:

| Class | Previous | Current | Delta |
|-------|---------|---------|-------|
| IMPORTANT | 92.2% | 92.2% | — |
| MODERATE | 44.4% | **48.1%** | +3.7pp |
| LOW | 60.0% | 60.0% | — |

IMPORTANT and LOW recall held steady while MODERATE recall picked up ~4 points. The precision jump (+4.5pp) is the most notable single-metric improvement. Given that the training data changed (fresh scrape), these gains are real but not guaranteed to be stable across future retrains.

## How to Test

```bash
# Or just run the test step against the already-trained model
make test-kernel
```

## Related Tickets
https://redhat.atlassian.net/browse/AEGIS-427

## Summary by Sourcery

Retrain the kernel CVE impact classifier with refreshed patch data and update all associated artifacts and evaluation outputs.

Enhancements:
- Refresh kernel CVE training and testing datasets, including SMOTE-balanced data and CVSS/CWE cache, from the latest upstream patch scrape.
- Update the kernel CVE impact XGBoost model and feature-importance outputs based on the new training corpus.
- Regenerate classifier evaluation artifacts, including prediction outputs and test summary JSON, from the retrained model.

### CODE REVIEW

**Review by kdudka — APPROVED** [review]
I like how the classification accuracy improves over time.  Please make sure the CI is green before merging this.
